tests/ttnn/unit_tests/operations/test_moreh_logsoftmax.py

```diff
@@ -400,3 +400,87 @@ def test_logsoftmax_backward_optional_output_tensor(shape_dim, optional_output_t
     passing, out = comp_allclose_and_pcc(x.grad, tt_dev, rtol=rtol, atol=atol)
     logger.info(out)
     assert passing
+
+
+@pytest.mark.parametrize(
+    "shape_dim",
+    (
+        ((50, 32), 1),  # single tile
+        ((3, 32, 32 * 5), 2),  # mutiple tile with dim W
+        ((5, 6, 32, 32), 3),  # multiple cores
+        ((10, 20, 32 * 3, 32 * 5), 3),  # multiple tiles per core
+        ((32, 32), 0),  # single tile
+        ((3, 32 * 5, 32), 1),  # mutiple tile with dim H
+        ((5, 6, 32, 32), 2),  # multiple cores
+        ((10, 20, 32 * 3, 32 * 5), 2),  # multiple tiles per core
+    ),
+)
+@pytest.mark.parametrize("compute_kernel_options", compute_kernel_options, ids=compute_kernel_ids)
+def test_callback_logsoftmax(shape_dim, compute_kernel_options, device, use_program_cache):
+    shape, dim = shape_dim
+    torch.manual_seed(0)
+
+    compute_kernel_config = get_compute_kernel_options(compute_kernel_options)
+
+    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16) + 100
+
+    dev_x = ttnn.Tensor(x, ttnn.bfloat16).pad_to_tile(float("nan")).to(ttnn.TILE_LAYOUT).to(device)
+
+    tt_cpu = F.log_softmax(x, dim)
+    for i in range(2):
+        tt_npu = ttnn.operations.moreh.logsoftmax(dev_x, dim, compute_kernel_config=compute_kernel_config)
+
+    tt_dev = tt_npu.cpu().to(ttnn.ROW_MAJOR_LAYOUT).unpad_from_tile(shape)
+    assert list(tt_dev.shape.with_tile_padding()) == list(tt_cpu.shape)
+    tt_dev = tt_dev.to_torch().to(torch.bfloat16)
+
+    rtol = atol = 0.1
+    passing, out = comp_allclose_and_pcc(tt_cpu, tt_dev, rtol=rtol, atol=atol)
+    logger.debug(out)
+    assert device.num_program_cache_entries() == 1
+    assert passing
+
+
+@pytest.mark.parametrize(
+    "shape_dim",
+    (
+        ((32, 32), 1),  # single tile
+        ((3, 32, 32 * 2), 2),  # mutiple tile with dim W
+        ((5, 6, 32, 32), 3),  # multiple cores
+        ((10, 20, 32 * 3, 32 * 5), 3),  # multiple tiles per core
+        ((32, 32), 0),  # single tile
+        ((3, 32 * 5, 32), 1),  # mutiple tile with dim H
+        ((5, 6, 32, 32), 2),  # multiple cores
+        ((10, 20, 32 * 5, 32), 2),  # multiple tiles per core
+    ),
+)
+@pytest.mark.parametrize("compute_kernel_options", compute_kernel_options, ids=compute_kernel_ids)
+def test_callback_logsoftmax_backward(shape_dim, compute_kernel_options, device):
+    device.enable_program_cache()
```

Comment:
should be replaced by `use_program_cache`.

Reply:
Resolved